spec/analytics_spec.coffee

```diff
@@ -32,37 +32,77 @@ describe 'Analytics', ->
 
   describe '.constructor', ->
     beforeEach ->
-      session = @session =
-        analytics_session: 'analytics_session'
-        shop_code: 'SA-XXXX-XXXX'
       @plugins_manager = new @PluginsManager()
-      @session_run_stub = sinon.stub(@Session::, 'run').returns(then: (fn) -> fn(session))
       @ActionsManager_spy::run = sinon.stub().returnsThis()
+      @_handle_session_spy = sinon.spy(@Analytics::, '_handle_session')
       @_live_spy = sinon.spy(@Analytics::, '_live')
-      @subject = new @Analytics()
 
     afterEach ->
-      @session_run_stub.restore()
+      @_handle_session_spy.restore()
       @_live_spy.restore()
 
-    it 'creates a PluginsManager using new', ->
-      expect(@PluginsManager).to.be.calledWithNew
+    context 'when a Session is acquired', ->
+      beforeEach ->
+        session = @session =
+          analytics_session: 'analytics_session'
+          shop_code: 'SA-XXXX-XXXX'
+        @session_run_stub = sinon.stub(@Session::, 'run').returns(then: (fn) -> fn(session))
+        @Settings.window.sa.q = [['category', 'command', 'param1', 'param2']]
+        @subject = new @Analytics()
 
-    it 'tries to acquire a Session', ->
-      expect(@session_run_stub).to.be.calledOnce
+      afterEach ->
+        @session_run_stub.restore()
 
-    context 'when a Session is acquired', ->
-      it 'creates an ActionsManager using new', ->
-        expect(@ActionsManager).to.be.calledWithNew
+      it 'creates a PluginsManager using new', ->
+        expect(@PluginsManager).to.be.calledWithNew
+
+      it 'tries to acquire a Session', ->
```

Comment:
The session is already acquired as implied from the context. Do you want to test if the `Session` object is initialized properly?

Reply:
I haven't changed this context. The bad context name is [this](https://github.com/skroutz/analytics.js/pull/82/files/91ee101025e500ac5f02b69644a2e42b28a38a94#diff-216799ae6dd5fb7a75ac24d1ccf6eacf1a338dcccf19db3c4c65a9a92426a9f3R44), I will change it to something that makes sense.